stubs/paho-mqtt/paho/mqtt/client.pyi

```diff
@@ -0,0 +1,321 @@
+    @property
+    def on_connect_fail(self) -> _OnConnectFail | None: ...
+    @on_connect_fail.setter
+    def on_connect_fail(self, func: _OnConnectFail) -> None: ...
+    def connect_fail_callback(self) -> Callable[[_OnConnectFail], _OnConnectFail]: ...
+    @property
+    def on_subscribe(self) -> _OnSubscribe | None: ...
+    @on_subscribe.setter
+    def on_subscribe(self, func: _OnSubscribe) -> None: ...
+    def subscribe_callback(self) -> Callable[[_OnSubscribe], _OnSubscribe]: ...
+    @property
+    def on_message(self) -> _OnMessage | None: ...
+    @on_message.setter
+    def on_message(self, func: _OnMessage) -> None: ...
+    def message_callback(self) -> Callable[[_OnMessage], _OnMessage]: ...
+    @property
+    def on_publish(self) -> _OnPublish | None: ...
+    @on_publish.setter
+    def on_publish(self, func: _OnPublish) -> None: ...
+    def publish_callback(self) -> Callable[[_OnPublish], _OnPublish]: ...
+    @property
+    def on_unsubscribe(self) -> _OnUnsubscribe | None: ...
+    @on_unsubscribe.setter
+    def on_unsubscribe(self, func: _OnUnsubscribe) -> None: ...
+    def unsubscribe_callback(self) -> Callable[[_OnUnsubscribe], _OnUnsubscribe]: ...
+    @property
+    def on_disconnect(self) -> _OnDisconnect: ...
+    @on_disconnect.setter
+    def on_disconnect(self, func: _OnDisconnect) -> None: ...
+    def disconnect_callback(self) -> Callable[[_OnDisconnect], _OnDisconnect]: ...
+    @property
+    def on_socket_open(self) -> _OnSocket: ...
+    @on_socket_open.setter
+    def on_socket_open(self, func: _OnSocket) -> None: ...
+    def socket_open_callback(self) -> Callable[[_OnSocket], _OnSocket]: ...
+    @property
+    def on_socket_close(self) -> _OnSocket: ...
+    @on_socket_close.setter
+    def on_socket_close(self, func: _OnSocket) -> None: ...
+    def socket_close_callback(self) -> Callable[[_OnSocket], _OnSocket]: ...
+    @property
+    def on_socket_register_write(self) -> _OnSocket: ...
+    @on_socket_register_write.setter
+    def on_socket_register_write(self, func: _OnSocket) -> None: ...
+    def socket_register_write_callback(self) -> Callable[[_OnSocket], _OnSocket]: ...
+    @property
+    def on_socket_unregister_write(self) -> _OnSocket: ...
+    @on_socket_unregister_write.setter
+    def on_socket_unregister_write(self, func: _OnSocket) -> None: ...
+    def socket_unregister_write_callback(self) -> Callable[[_OnSocket], _OnSocket]: ...
+    def message_callback_add(self, sub: str, callback: _OnMessage) -> None: ...
+    def topic_callback(self, sub: str) -> Callable[[_OnMessage], _OnMessage]: ...
+    def message_callback_remove(self, sub: str) -> None: ...
+
+class WebsocketWrapper:
+    OPCODE_CONTINUATION: int
+    OPCODE_TEXT: int
+    OPCODE_BINARY: int
+    OPCODE_CONNCLOSE: int
+    OPCODE_PING: int
+    OPCODE_PONG: int
+    connected: bool
+    def __init__(self, socket: Any, host: str, port: int, is_ssl: bool, path: str, extra_headers: _ExtraHeader) -> None: ...
```

Comment:
We should probably use either `socket.socket` or `_typeshed.Incomplete` here instead of `Any` for `socket`.

Reply:
Sounds good.

Reply:
The `socket` parameter can be `socket.socket | ssl.SSLSocket | socks.socksocket`.
But I use `Any` instead of `socks.socksocket` because `pysocks` is not required for `paho-mqtt`.
Or should I include `pysocks` to the list of `requires` in `METADAT.toml` for completeness?

Reply:
I think the best solution would be to define a protocol. But for now maybe just annotate it with `socket.socket | ssl.SSLSocket | Any` and add a comment.